File: UserDev/ClusterStudy/mac/test_merge.py
import ROOT,sys
ROOT.gSystem.Load("libClusterRecoUtil")
from ROOT import *
from ROOT import larlight as fmwk

mgr = fmwk.ana_processor()

mgr.add_input_file(sys.argv[1])

mgr.set_input_rootdir("scanner")

mgr.set_output_file("")

mgr.set_io_mode(fmwk.storage_manager.READ)

mgr.set_ana_output_file("")

proc = fmwk.ClusterMerger()

proc.GetManager().AddMergeAlgo(cluster.CMAlgoFake())

proc.GetManager().DebugMode(cluster.CMergeManager.kPerIteration)
#proc.GetManager().DebugMode(cluster.CMergeManager.kPerAlgoSet)
#proc.GetManager().DebugMode(cluster.CMergeManager.kPerMerging)

mgr.add_process(proc)

while mgr.process_event():

    print 
    bk=proc.GetManager().GetBookKeeper()

    res = bk.GetResult()

    for x in xrange(res.size()):

        for y in xrange(res.at(x).size()):

            print res.at(x).at(y),
        print

    sys.stdin.readline()

File: UserDev/ClusterStudy/bin/test_merge.cc
#include "Analysis-TypeDef.hh"
#include "ClusterStudy-TypeDef.hh"
#include "TSystem.h"

int main(int argc, char** argv) {

  larlight::ana_processor mgr;

  mgr.add_input_file("../../../combined_02.root");
    
  mgr.set_input_rootdir("scanner");

  mgr.set_output_file("");

  mgr.set_io_mode(larlight::storage_manager::READ);
  
  mgr.set_ana_output_file("");
  
  larlight::ClusterMerger* proc = new larlight::ClusterMerger;;

  proc->GetManager().AddMergeAlgo(new cluster::CMAlgoFake());

  proc->GetManager().AddSeparateAlgo(new cluster::CMAlgoFake());

  proc->GetManager().DebugMode(cluster::CMergeManager::kPerIteration);
  //proc.GetManager().DebugMode(cluster.CMergeManager.kPerAlgoSet);
  //proc.GetManager().DebugMode(cluster.CMergeManager.kPerMerging);

  mgr.add_process(proc);

  while (1) {
    
    mgr.process_event();

  };

  return 1;
}



